### ISSUE TITLE
Gold Price takes card out of hand if Adam Marbrand takes character back to hand as first player

### ISSUE THREAD

**Issue by hkroeger1806** [throneteki/throneteki#3503]
If you are first player and Sir Adam Marbrand takes a discarded character back to hand, the Gold Price can be triggered and put the character from your hand into play which should not be allowed.

**Comment by Patane97** [comment]
@hkroeger1806 Could we just get a bit more information here? Listing the order of actions that takes place to reproduce this issue would help us figure out the underlying problem. :)

**Comment by hkroeger1806** [comment]
Hi, yes sorry, that was a bit short:

1. First player has Ser Adam Marbrand on the board with at least 1 gold
2. Second player plays The Gold Price
3. Second player makes an intrigue challenge and wins.
4. A character is discarded
5. Ser Adam Marbrand is triggered by the first player in their reaction window to a character being discarded
6. The character returns to first player' hand
7. The Second player triggers The Gold Price to the character being discarded
8. The character vanishes from first player's hand and is put into play under the second player's control.

If you need more, please let me know and many thanks!

**Comment by Patane97** [comment]
> Hi, yes sorry, that was a bit short and useless:
> 
> 1. First player has Ser Adam Marbrand on the board with at least 1 gold
> 2. Second player plays The Gold Price
> 3. Second player makes an intrigue challenge and wins.
> 4. A character is discarded from first player's hand
> 5. Ser Adam Marbrand is triggered by the first player in their reaction window to a character being discarded
> 6. The character returns to first player's hand
> 7. The Second player triggers The Gold Price to the character being discarded
> 8. The character vanishes from first player's hand and is put into play under the second player's control.
> 
> If you need more, please let me know and many thanks!

Nope, this is perfect! Makes more sense, totally get what's happening now.

I might flag this one to not fix anytime soon, as there is a bigger update I'm working on which should fix a bunch of small bugs like this - primarily when there is one effect (eg. Ser Adam Marbrand) that invalidates another effect from being triggerable (eg. Gold Price, as the target card is no longer visible).

Will reference this issue once that larger update is in, and ensure it fixes this one. :) Thanks for the report & summary, really helps 💪
